src/Confluent.Kafka/Producer.cs

```diff
@@ -194,6 +194,150 @@ public void HandleDeliveryReport(Message message)
                 SetResult(message);
             }
         }
+        
+        /// <summary>
+        ///     Detailed messaged
+        /// </summary>
+        public struct ProduceRecord
```

Comment:
nested classes are typically frowned upon I believe (I think this should be moved out and into a new file).